code/game/objects/structures/watercloset.dm

```diff
@@ -355,24 +362,31 @@
 			var/mob/living/carbon/C = O
 			check_heat(C)
 		wash(O)
-		O.clean_blood()
 		watersource.reagents.reaction(O, TOUCH)
 		if(istype(O, /obj/item/weapon/reagent_containers/glass))
 			var/obj/item/weapon/reagent_containers/glass/G = O
 			G.reagents.add_reagent("water", 5)
 	watersource.reagents.reaction(get_turf(src), TOUCH)
 
 /obj/machinery/shower/proc/check_heat(mob/living/carbon/C as mob)
-	if(!on || watertemp == "cool")
+	if(!on)
 		return
 
 	//Note : Remember process() rechecks this, so the mix/max procs slowly increase/decrease body temperature
-	if(watertemp == "freezing cold")
-		C.bodytemperature = max(T0C - 10, C.bodytemperature - 0.5) //Down to -10�C - sorry.
+	//Every second under the shower adjusts body temperature by 0.5�C. Water conducts heat pretty efficiently in real life too
+	if(watertemp == "freezing cold") //Down to 0�C, Nanotrasen waterworks are perfect and never fluctuate even slightly below that
+		C.bodytemperature = max(T0C, C.bodytemperature - 0.5)
 		return
-	if(watertemp == "searing hot")
-		C.bodytemperature = min(T0C + 300, C.bodytemperature + 50) //Up to 300�C
+	if(watertemp == "searing hot") //Up to 60�c, upper limit for common water boilers
+		C.bodytemperature = min(T0C + 60, C.bodytemperature + 0.5)
 		return
+	if(watertemp == "cool") //Adjusts towards "perfect" body temperature, 37.5�C. Actual showers tend to average at 40�C, but it's the future
+		if(C.bodytemperature > T0C + 37.5) //Cooling down
+			C.bodytemperature = min(T0C + 37.5, C.bodytemperature - 0.5)
```

Comment:
I think you meant "max", this makes your temperature instantly jump from 150Cº to 37.5Cº :^)